contracts/sg721-base/src/contract.rs

```diff
@@ -214,6 +221,82 @@ where
         Ok(res)
     }
 
+    pub fn update_collection_info(
+        &self,
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        new_collection: CollectionInfo<RoyaltyInfoResponse>,
+    ) -> Result<Response, ContractError> {
+        let frozen_collection_info = self.frozen_collection_info.load(deps.storage)?;
+
+        if frozen_collection_info {
+            return Err(ContractError::CollectionInfoFrozen {});
+        }
+
+        let collection = self.query_collection_info(deps.as_ref())?;
+        if collection.creator != info.sender {
+            return Err(ContractError::Unauthorized {});
+        }
+
+        deps.api.addr_validate(&new_collection.creator)?;
```

Comment:
do we really want to change the creator? nothing stops this from impersonating another creator.